ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/TagsDemoActivity.kt

```diff
@@ -0,0 +1,226 @@
+
+        }
+    }
+
+    private fun addObjectsToMap() {
+        with(map) {
+
+            // A circle centered on Adelaide.
+            addCircle(CircleOptions().apply {
+                center(places.getValue("ADELAIDE"))
+                radius(500000.0)
+                fillColor(Color.argb(150, 66, 173, 244))
+                strokeColor(Color.rgb(66, 173, 244))
+                clickable(true)
+            }).run {
+                // add a tag to the circle to count clicks
+                tag = CustomTag("Adelaide circle")
+            }
+
+            // A ground overlay at Sydney.
+            addGroundOverlay(GroundOverlayOptions().apply {
+                image(BitmapDescriptorFactory.fromResource(R.drawable.harbour_bridge))
+                position(places.getValue("SYDNEY"), 700000f)
+                clickable(true)
+            }).run {
+                // add a tag to the overlay to count clicks
+                tag = CustomTag("Sydney ground overlay")
+            }
+
+            // A marker at Hobart.
+            addMarker(MarkerOptions().apply {
+                position(places.getValue("HOBART"))
+            }).run {
+                // add a tag to the marker to count clicks
+                tag = CustomTag("Hobart marker")
+            }
+
+            // A polygon centered at Darwin.
+            addPolygon(PolygonOptions().apply{
+                add(LatLng(places.getValue("DARWIN").latitude + 3,
+                        places.getValue("DARWIN").longitude - 3),
+                        LatLng(places.getValue("DARWIN").latitude + 3,
+                                places.getValue("DARWIN").longitude + 3),
+                        LatLng(places.getValue("DARWIN").latitude - 3,
+                                places.getValue("DARWIN").longitude + 3),
+                        LatLng(places.getValue("DARWIN").latitude - 3,
+                                places.getValue("DARWIN").longitude - 3))
+                fillColor(Color.argb(150, 34, 173, 24))
+                strokeColor(Color.rgb(34, 173, 24))
+                clickable(true)
+            }).run {
+                // add a tag to the marker to count clicks
+                tag = CustomTag("Darwin polygon")
+            }
+
+            // A polyline from Perth to Brisbane.
+            addPolyline(PolylineOptions().apply{
+                add(places.getValue("PERTH"), places.getValue("BRISBANE"))
+                color(Color.rgb(103, 24, 173))
+                width(30f)
+                clickable(true)
+            }).run {
+                // add a tag to the polyline to count clicks
+                tag = CustomTag("Perth to Brisbane polyline")
+            }
+        }
+    }
+
+    // Click event listeners.
+    private fun onClick(tag: CustomTag) {
+        tag.incrementClickCount()
+        tagText.text = tag.toString()
+    }
+
+    override fun onCircleClick(circle: Circle) {
+        onClick(circle.tag as CustomTag)
```

Comment:
Do you need to handle the case when the cast fails?

Reply:
Yep! I've made it so the function will just return if the cast cannot be completed.